File: backend/models/AppSetting.js
const mongoose = require('mongoose');

const settingSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true,
    index: true
  },
  settingKey: {
    type: String,
    required: true
  },
  value: mongoose.Schema.Types.Mixed,
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

settingSchema.index({ settingKey: 1 }, { unique: false });

module.exports = mongoose.model('AppSetting', settingSchema);

File: lib/models/app_setting.dart
class AppSetting {
  final String? id;
  final String userId;
  final String settingKey;
  final dynamic value;
  final DateTime updatedAt;

  AppSetting({
    this.id,
    required this.userId,
    required this.settingKey,
    required this.value,
    required this.updatedAt,
  });

  Map<String, dynamic> toJson() => {
    if (id != null) '_id': id,
    'userId': userId,
    'settingKey': settingKey,
    'value': value,
    'updatedAt': updatedAt.toIso8601String(),
  };

  factory AppSetting.fromJson(Map<String, dynamic> json) => AppSetting(
    id: json['_id']?.toString(),
    userId: json['userId'] ?? '',
    settingKey: json['settingKey'] ?? '',
    value: json['value'],
    updatedAt: json['updatedAt'] != null 
        ? DateTime.parse(json['updatedAt']) 
        : DateTime.now(),
  );
}